packages/guides/src/Nodes/SectionNode.php

```diff
@@ -33,4 +35,19 @@ public function getTitles(): array
 
         return $titles;
     }
+
+    public function getLinkType(): string
+    {
+        return self::STD_LABEL;
+    }
+
+    public function getId(): string
+    {
+        return $this->getTitle()->getId();
+    }
+
+    public function getLinkText(): string
+    {
+        return $this->getTitle()->toString();
+    }
```

Comment:
Is this not rendering information? A SectionNode can be rendered as a "LinkTarget" or as a "Section", in the first case we use a specific rendering strategy to transform the SectionNode to a link. 

This will make the SectionNode unaware of the contexts using the node. The node itself doesn't have to be aware of this. Which makes it easier to maintain this node or introduce new nodes with the same way of rendering.

Reply:
What I was doing here was introducing a "LinkTargetNode"  interface that can be implemented by any node that can be linked to. Now anything that can be linked to supplys a default link text that can be used when no explicit linktext is supplied by the link. This link text is not used for rendering the node itself but any link to it. By making also the Section Node implement the interface the section node itself can be treated like any other linkable object.